### PR TITLE
Feature/calendar theme

### PR DESCRIPTION
**Pull request by mattias800**
New styling system, where theme contains functions, instead of just CSS objects.
New highlight system, where each day can have an array of highlights. Theme functions receive these as arguments.
Add support for custom content in header of calendar (left and right of month name).
Add new month switcher with buttons in header of calendar.
Add support for custom content in day cells.
Add new theme colors.
Add new stories.
Add theme for extranet.

### CODE REVIEW

**Comment by knegusen** [comment]
image
image
image

File: src/components/ui/form/calendar/components/Calendar.tsx [hunk 1]
```diff
@@ -25,12 +22,25 @@ import { CalendarDay } from './renderers/CalendarDay';
 
 export type __C13581358 = InferableComponentEnhancerWithProps<{}, {}>;
 
-export interface CalendarDayProps<T = {}> {
+/*
```
Comment (knegusen) [inline src/components/ui/form/calendar/components/Calendar.tsx]: Should this be here? What's done? what's left?

File: src/components/ui/form/calendar/components/CalendarTheme.tsx [hunk 2]
```diff
@@ -26,27 +27,158 @@ export interface WeekDayTheme {
...
+    textProps: defaultTextPropsProvider({
+      selectedColor: defaultColors.white,
+      disabledColor: defaultColors.disabledText,
+      inOtherMonthColor: 'transparent',
+    }),
+  },
+  CalendarMonth: {},
+};
+
+export const extranetCalendarTheme: CalendarTheme = {
```
Comment (knegusen) [inline src/components/ui/form/calendar/components/CalendarTheme.tsx]: We should move this constant to the extranet project instead?

Comment (mattias800) [reply src/components/ui/form/calendar/components/CalendarTheme.tsx]: I'll leave it, maybe move it in another PR?

**Pull request merged**